packages/cli/src/swagger/specGenerator3.ts

```diff
@@ -605,14 +598,14 @@ export class SpecGenerator3 extends SpecGenerator {
   // duplicate types.
   private groupEnums(types: Array<Swagger.Schema | Swagger.BaseSchema>) {
     const returnTypes: Array<Swagger.Schema | Swagger.BaseSchema> = [];
-    const enumValuesByType = {};
+    const enumValuesByType: Record<string, Record<string, boolean | string | number | null>> = {};
     for (const type of types) {
       if (type.enum && type.type) {
         for (const enumValue of type.enum) {
           if (!enumValuesByType[type.type]) {
-            enumValuesByType[type.type] = [];
+            enumValuesByType[type.type] = {};
```

Comment:
Can we add a test here? It seems like this should be covered.

Reply:
The result of the `groupEnums` should be the same, as JS allows setting string keys even on arrays. Thus, the array behaves as an object:
<img width="183" alt="image" src="https://github.com/lukeautry/tsoa/assets/14100405/6f74dc5c-2377-470a-9db4-44dd04e0bd0a">

How would I test this change, any tips?

Reply:
I see why this change makes sense, should've always been an `{}`.